src/symbolic/symbolic_memory_concretizing.ml

```diff
@@ -108,15 +108,13 @@ module Backend = struct
     match Smtml.Expr.view a with
     | Val (Num (I32 _)) -> return (Ok a) (* An i32 is a valid address *)
     | Ptr { base; offset } -> (
-      let (* A pointer is valid if it's within bounds. *)
-      open Symbolic_value in
+      let open Symbolic_value in
+      (* A pointer is valid if it's within bounds. *)
       match Hashtbl.find m.chunks base with
       | exception Not_found -> return (Error Trap.Memory_leak_use_after_free)
       | size ->
-        let base = const_i32 base in
-        let ptr = I32.add base offset in
         let+ is_out_of_bounds =
-          select (Bool.or_ (I32.lt ptr base) (I32.ge ptr (I32.add base size)))
+          select (Bool.or_ (I32.lt offset (const_i32 0l)) (I32.ge offset size))
```

Comment:
Shouldn't the offset be interpreted as an unsigned int ? It would remove the need for testing positivity here.

Reply:
Indeed, thanks!